internal/provider/resources/branch/resource.go

```diff
@@ -0,0 +1,197 @@
+	base abstraction.BaseResource[Model, *keboola.Branch]
+
+	// Direct access to the API client for specific operations
+	client    *keboola.AuthorizedAPI
+	projectID int
+}
+
+// NewResource is a helper function to simplify the provider implementation.
+func NewResource() *Resource {
+	return &Resource{}
+}
+
+// Metadata returns the resource type name.
+func (r *Resource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_branch"
+}
+
+// Schema defines the schema for the resource.
+func (r *Resource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server
+		MarkdownDescription: "Branch resource",
+		Description:         "Development branch resource",
+		DeprecationMessage:  "",
+		Version:             1,
+		Blocks:              map[string]schema.Block{},
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				MarkdownDescription: "Branch ID",
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Branch name",
+				Required:            true,
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "Branch description",
+				Optional:            true,
+				Default:             stringdefault.StaticString(""),
+				Computed:            true,
+			},
+			"is_default": schema.BoolAttribute{
+				MarkdownDescription: "Default flag",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+// Configure adds the provider configured client to the resource.
+func (r *Resource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	// Return silently if provider data is not available (yet)
+	if req.ProviderData == nil {
+		return
+	}
+
+	// Get the provider data - ignoring the type assertion success
+	providerData, _ := req.ProviderData.(*providermodels.ProviderData)
+
+	// Set up the API client
+	r.client = providerData.Client
+	r.projectID = providerData.Token.ProjectID()
+
+	// Set up the mapper
+	r.base.Mapper = &Mapper{
+		projectID: r.projectID,
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "Creating branch resource")
+
+	// Use the base resource abstraction for Create
+	r.base.ExecuteCreate(ctx, req, resp, func(ctx context.Context, model Model) (*keboola.Branch, error) {
+		// Handle API call from the mapper
+		emptyModel := Model{
+			ID:          types.Int64Null(),
+			Name:        types.StringNull(),
+			Description: types.StringNull(),
+			IsDefault:   types.BoolNull(),
+		}
+
+		// Map the Terraform model to the API model
+		apiModel, err := r.base.Mapper.MapTerraformToAPI(ctx, emptyModel, model)
```

Comment:
```suggestion
		// Map the Terraform model to the API model
		apiModel, err := r.base.Mapper.MapTerraformToAPI(ctx, Model{}, model)
```
Do we need to create empty Model with default types? :thinking:

Reply:
Not sure, I copied that.

Reply:
so fix it? Document it? The code should be aligned between each resource as much as can, because there is always `tfstate` -> `API` and `API` -> `tfstate` based on schema. That's why I created mapper and methods that are updating `tfstate`.